netket/optimizer/qgt/qgt_jacobian_common.py

```diff
@@ -111,7 +112,55 @@ def choose_jacobian_mode(afun, pars, state, samples, *, mode, holomorphic):
         raise ValueError(f"unknown mode {i}")
 
 
-def rescale(centered_oks, *, ndims: int = 1):
+def sanitize_diag_shift(diag_shift, diag_scale, rescale_shift):
+    """Sanitises different inputs for diag_shift etc.
+
+
+    Also raises a deprecation warnings for `rescale_shift`.
+
+    Returns:
+        the tuple `(diag_shift, diag_scale)`.
+    """
+
+    if diag_shift is None:
+        diag_shift = 0.01 if diag_scale is None else 0.0
+
+    if rescale_shift is False:
+        warn_deprecation(
+            "`rescale_shift` is deprecated, please do not specify `rescale_shift=False`."
+        )
+        if diag_scale is not None:
+            raise ValueError(
+                "`rescale_shift` and `diag_scale` must not be specified together."
+            )
+
+        return diag_shift, 0.0
+    elif rescale_shift is True:
+        warn_deprecation(
+            f"`rescale_shift` is deprecated, use `diag_scale={diag_shift}, diag_shift=0` instead."
+        )
+        if diag_scale is not None:
+            raise ValueError(
+                "`rescale_shift` and `diag_scale` must not be specified together."
+            )
+
+        return 0.0, diag_shift
+    elif rescale_shift is None:
+        if diag_scale is None:
+            diag_scale = 0.0
```

Comment:
I'm mainly advocating for removing those two lines

Reply:
that's the opposite of `sanitizing` the inputs. currently, `sanitize...` turns any input into two numbers, so `to_scale_offset` doesn't have to worry about all the possibilities.

Reply:
If it was complicated to support both, I'd agree with you. But as our QGT logic already supports both cases, I don't see why an intermediate function (which would literally not even add another line) should not support it.